docs/docs/testing-your-assistant.mdx

```diff
@@ -377,6 +377,26 @@ The test script will also save a confusion matrix to a file called
 matrix shows how often the action was correctly predicted and how often an
 incorrect action was predicted instead.
 
+### Interpreting the generated warnings
+
+The test script will also generate a warnings file called `results/stories_with_warnings.yml`.
+This file contains all test stories for which [`action_unlikely_intent`](./default-actions.mdx#action_unlikely_intent)
+was predicted at any conversation turn but all actions from the original story were predicted correctly.
+However, if a test story originally consisted of `action_unlikely_intent` but wasn't predicted by
+the ensemble of policies, then the corresponding story will end up in `results/failed_test_stories.yml` as
+a failed story.
```

Comment:
This is a little confusingly worded. What about:
However, if a test story originally included an `action_unlikely_intent` ([as part of a rule](link to customizing unlikely intents) for example) but wasn't predicted by
the `IntentTEDPolicy`, then the corresponding story will end up in `results/failed_test_stories.yml` as
a failed story.

Also, I changed ensemble of policies to IntentTEDPolicy above since only this one can predict action unlikely intent right?

Reply:
Not sure I completely agree here - 

> However, if a test story originally included an action_unlikely_intent ([as part of a rule]

Whether it's part of a rule or not does not matter because this is a story and rule snippets are not part of a story.

> Also, I changed ensemble of policies to IntentTEDPolicy above since only this one can predict action unlikely intent right?

I think it should be ensemble of policies (or dialogue manager) because it does not matter whether a particular policy predicted it or not. The test story is evaluated against what the ensemble predicts. For example, it can happen that the ensemble didn't choose `action_unlikely_intent` because a rule was also matching for some reason and `RulePolicy` predicted an alternate action.

Reply:
I see both points. Re: ensemble vs IntentTEDPolicy, your explanation makes sense and I think keeping the ensemble of policies is fine. 

Regarding (as part of a rule..), this can be deleted but I think what I am worried about is it may not be clear at this point why you would include an `action_unlikely_intent` in a test story, and there could be some brief motivation or pointer to motivation about why you would do that. What I was hoping to draw reference to with the mention of the rule was the scenario where you included a rule within a larger test story that tested both that the `action_unlikely_intent` got triggered where you expected and then the corresponding rule got triggered.

Reply:
>  What I was hoping to draw reference to with the mention of the rule was the scenario where you included a rule within a larger test story that tested both that the action_unlikely_intent got triggered where you expected and then the corresponding rule got triggered. 

This makes sense. i reworded this to:

```
However, if a test story originally included an `action_unlikely_intent`, for example to ensure [a rule triggers what
happens after an `action_unlikely_intent`](./default-actions.mdx#customization-1) but the ensemble of
policies failed to do so, then the corresponding story will end up in `results/failed_test_stories.yml` as
a failed story.
```
Wdyt?

Reply:
@kedz Could you please take a look at this again?